src/examples/generation.ts

```diff
@@ -55,40 +59,74 @@ const bone = Armature.define((root: Node) => {
 
 const treeGen = Armature.generator();
 treeGen
-    .define('branch', (root: Point) => {
-        const node = bone();
+    .define('branch', (root: Point, instance: GeneratorInstance) => {
+        const node = instance.add(bone());
         node.point('base').stickTo(root);
+        node.scale(Math.random() * 0.4 + 0.9);
         node
             .hold(node.point('tip'))
             .rotate(Math.random() * 360)
             .release();
         node
             .hold(node.point('handle'))
-            .rotate(Math.random() * 45)
+            .rotate(Math.random() * 70)
             .release();
         node.scale(0.8); // Shrink a bit
 
-        const trunk = node.point('mid').attach(branchShape);
+        const trunk = instance.add(node.point('mid').attach(branchShape));
         trunk.scale({ x: 0.2, y: 1, z: 0.2 });
 
-        treeGen.addDetail({ component: 'branchOrLeaf', at: node.point('tip') });
+        instance.addDetail({ component: 'branchOrLeaf', at: node.point('tip') });
     })
-    .defineWeighted('branchOrLeaf', 1, (root: Point) => {
-        treeGen.addDetail({ component: 'leaf', at: root });
+    .defineWeighted('branchOrLeaf', 1, (root: Point, instance: GeneratorInstance) => {
+        instance.addDetail({ component: 'leaf', at: root });
     })
-    .defineWeighted('branchOrLeaf', 4, (root: Point) => {
-        treeGen.addDetail({ component: 'branch', at: root });
-        treeGen.addDetail({ component: 'maybeBranch', at: root });
-        treeGen.addDetail({ component: 'maybeBranch', at: root });
+    .defineWeighted('branchOrLeaf', 4, (root: Point, instance: GeneratorInstance) => {
+        instance.addDetail({ component: 'branch', at: root });
+        instance.addDetail({ component: 'maybeBranch', at: root });
+        instance.addDetail({ component: 'maybeBranch', at: root });
     })
-    .define('leaf', (root: Point) => {
-        const leaf = root.attach(leafSphere);
+    .define('leaf', (root: Point, instance: GeneratorInstance) => {
+        const leaf = instance.add(root.attach(leafSphere));
         leaf.scale(Math.random() * 0.5 + 0.5);
     })
-    .maybe('maybeBranch', (root: Point) => {
-        treeGen.addDetail({ component: 'branch', at: root });
+    .maybe('maybeBranch', (root: Point, instance: GeneratorInstance) => {
+        instance.addDetail({ component: 'branch', at: root });
     });
-const tree = treeGen.generate({ start: 'branch', depth: 25 });
+
+/*const tree = treeGen.generateSOSMC({
```

Comment:
Do you want this comment to stay here?

Reply:
Leaving it for now as an example of how to use the attractor/repulsor system in case anyone wants to try it